tools/spack-scripting/scripting/cmd/devbuildcosmo.py

```diff
@@ -82,12 +82,19 @@ def devbuildcosmo(self, args):
 
     # Set dycore_spec
     if not args.without_dycore:
-        dycore_spec = 'cosmo-dycore@dev-build'
+        dycore_spec = 'cosmo-dycore@dev-build '
     else:
-        dycore_spec = 'cosmo-dycore@master'
-    dycore_spec += ' real_type=' + cosmo_spec.variants['real_type'].value
+        dycore_spec = 'cosmo-dycore@master '
+
+    # extracting dycore variants
+    dycore_spec += cosmo_spec.format('{^cosmo-dycore.variants}')
+
+    # extracting correct mpi variant
     dycore_spec += ' ^' + cosmo_spec.format('{^mpi.name}') + '%' + cosmo_spec.compiler.name
 
+    # remove the slum_args variant causing troubles to the concretizer
```

Comment:
typo in comments: slum_args
```suggestion
    # remove the slurm_args variant causing troubles to the concretizer
```